azure-pipelines.yml

```diff
@@ -4,159 +4,551 @@ variables:
+                    pnpm run setup
+                  workingDirectory: '$(AppHostingSdkProjectDirectory)'
+              - task: Bash@3
+                displayName: 'Run E2E Perf tests'
+                condition: succeeded()
+                inputs:
+                  targetType: inline
+                  script: 'node tools/cli/runAppsWithE2ETests.js --appUrl=https://localhost:4002 --reportFileName=e2e-tests-report-perf --envType=perf'
+                  workingDirectory: '$(AppHostingSdkProjectDirectory)'
+              - task: PublishTestResults@2
+                inputs:
+                  testResultsFormat: 'JUnit'
+                  testResultsFiles: '**/e2e-tests-report*.xml'
+                  testRunTitle: 'E2E Tests - Perf'
+                  mergeTestResults: true
+                condition: succeededOrFailed()
+          - job: E2ETestAndroid
+            displayName: 'E2E Test - Android'
+            pool:
+              name: Azure Pipelines
+              image: internal-macos12
+              os: macOS
+            steps:
+              - task: 6d15af64-176c-496d-b583-fd2ae21d4df4@1
+                inputs:
+                  repository: self
+              - task: 6d15af64-176c-496d-b583-fd2ae21d4df4@1
+                inputs:
+                  repository: AndroidAppHostingSdk
+                  path: androidHost
+                  persistCredentials: true
+              - task: NodeTool@0
+                inputs:
+                  versionSpec: '14.x'
+                displayName: 'Install Node.js'
+              - task: Cache@2
+                inputs:
+                  key: 'pnpm | "$(Agent.OS)" | $(ClientSdkProjectDirectory)/pnpm-lock.yaml'
+                  path: $(Pipeline.Workspace)/.pnpm-store
+                displayName: Cache pnpm
+              - task: CmdLine@2
+                displayName: 'Setup pnpm'
+                inputs:
+                  script: |
+                    corepack enable
+                    corepack prepare pnpm@7.30.1 --activate
+                    pnpm config set store-dir $(Pipeline.Workspace)/.pnpm-store
+              - task: CmdLine@2
+                displayName: 'Install client sdk dependencies'
+                inputs:
+                  script: |
+                    pnpm install
+                  workingDirectory: '$(ClientSdkProjectDirectory)'
+              - task: CmdLine@2
+                displayName: 'Build client sdk'
+                inputs:
+                  script: |
+                    pnpm build-force-blazor
+                  workingDirectory: '$(ClientSdkProjectDirectory)'
+              - task: Bash@3
+                displayName: 'Run sample test app in background'
+                inputs:
+                  targetType: inline
+                  script: 'nohup pnpm start-test-app &'
+                  workingDirectory: '$(ClientSdkProjectDirectory)'
+              - task: Bash@3
+                displayName: 'Install Emulator'
```

Comment:
Can you copy over the 10 minute timeout on this task? If it goes longer than that, it almost always times out after the whole 60 minutes.